src/providers/azure.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * Azure search index provider.
+ */
+class AzureIndex {
+  constructor(apiKey, serviceName, name, downloader) {
+    this._apiKey = apiKey;
+    this._serviceName = serviceName;
+    this._name = name;
+    this._downloader = downloader;
+  }
+
+  async search(query, { hitsPerPage, page, attributesToRetrieve }) {
+    const url = new URL(`https://${this._serviceName}.search.windows.net/indexes/${this._name}/docs`);
+    url.searchParams.append('api-version', '2019-05-06');
+    url.searchParams.append('$top', hitsPerPage);
+    url.searchParams.append('$skip', page * hitsPerPage);
+    url.searchParams.append('$select', attributesToRetrieve.map((k) => k.replace(/-/g, '_')).join(','));
```

Comment:
Azure doesn't allow `-` in column names. We therefore convert them to underscores